### PR TITLE
Allowing user to invalidate paid invoice

### PR DESCRIPTION
**Pull request by lepipele**

### CODE REVIEW

**Comment by NicolasDorier** [comment]
Can you rebase ?

File: BTCPayServer/Services/Invoices/InvoiceRepository.cs [hunk 1]
```diff
@@ -251,6 +251,18 @@ void AddToTextSearch(string invoiceId, params string[] terms)
             }
         }
 
+        public async Task UpdatePaidInvoiceToInvalid(string invoiceId)
+        {
+            using (var context = _ContextFactory.CreateContext())
+            {
+                var invoiceData = await context.FindAsync<InvoiceData>(invoiceId).ConfigureAwait(false);
+                if (invoiceData == null || invoiceData.Status != "paid")
```
Comment (NicolasDorier) [inline BTCPayServer/Services/Invoices/InvoiceRepository.cs]: nit: Can be simplified `invoiceData?Status != null`

**Comment by lepipele** [comment]
Merged

**Comment by lepipele** [comment]
Merged @NicolasDorier

**Pull request merged**